tests/verifications/generate_report.py

```diff
@@ -7,24 +7,40 @@
 """
 Test Report Generator
 
+Description:
+    This script runs pytest tests (specifically designed for OpenAI API compatibility checks)
+    for different providers, aggregates the results from JSON reports, and generates
+    a markdown summary report (REPORT.md).
+
+    It automatically cleans up old test result files, keeping only the latest
+    per provider.
+
 Requirements:
-    pip install pytest-json-report
+    pip install pytest-json-report PyYAML
```

Comment:
can we use https://docs.astral.sh/uv/guides/scripts/#declaring-script-dependencies